src/store/review-context.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+
+import axios from 'axios';
+
+import { API } from '../config';
+
+type Review = {
+  title: string;
+  category: string;
+  designer: string;
+  owner: string;
+  review_img_url: string;
+  created_at: Date;
+  votes: number;
+  comment_count: number;
+};
+
+export type ContextDefaults = {
+  reviews: Array<Review>;
+};
+
+const contextDefaults = {
+  reviews: [],
+};
+
+export const ReviewContext =
+  React.createContext<ContextDefaults>(contextDefaults);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const ReviewContextProvider = ({ children }: Props) => {
+  const [reviews, setReview] = useState([]);
+
+  useEffect(() => {
+    const getAllReviews = async () => {
+      const reviews = await axios.get(`${API}/reviews`);
+
+      setReview(reviews.data.reviews);
+    };
+
+    getAllReviews();
+  }, []);
+
+  const contextValue = {
+    reviews,
+  };
+
+  return (
+    <ReviewContext.Provider value={contextValue}>
+      {children}
+    </ReviewContext.Provider>
+  );
+};
```

Comment:
Its great you've looks into useContext. Only issue would be with the implementation - you've made global state for something that only gets used in one place (at the moment). This would be better suited to a customHook, which does the useEffect and handles the state - then its invoked when needed somewhere else

Reply:
The other issue with this is it will always do this api request when the app loads, if someone never goes straight to a users page, potentially an operation happening that doesnt need to be